pkg/validation-webhook/snapshot.go

```diff
@@ -223,6 +242,51 @@ func decideSnapshotContentV1(snapcontent, oldSnapcontent *volumesnapshotv1.Volum
 	return reviewResponse
 }
 
+func decideSnapshotClassV1beta1(snapClass, oldSnapClass *volumesnapshotv1beta1.VolumeSnapshotClass, lister storagelisters.VolumeSnapshotClassLister) *v1.AdmissionResponse {
+	reviewResponse := &v1.AdmissionResponse{
+		Allowed: true,
+		Result:  &metav1.Status{},
+	}
+
+	// Only Validate when a new snapClass is being set as a default.
+	if snapClass.Annotations[utils.IsDefaultSnapshotClassAnnotation] != "true" {
+		return reviewResponse
+	}
+
+	// If Old snapshot class has this, then we can assume that it was validated
+	if oldSnapClass.Annotations[utils.IsDefaultSnapshotClassAnnotation] == "true" {
+		return reviewResponse
+	}
+
+	ret, err := lister.List(labels.Everything())
+	if err != nil {
+		reviewResponse.Allowed = false
+		reviewResponse.Result.Message = err.Error()
+		return reviewResponse
+	}
+
+	driverToSnapshotDefautlMap := map[string]*volumesnapshotv1.VolumeSnapshotClass{}
+	for _, snapshotClass := range ret {
+		if snapshotClass.Annotations[utils.IsDefaultSnapshotClassAnnotation] != "true" {
+			continue
+		}
+		if _, ok := driverToSnapshotDefautlMap[snapshotClass.Driver]; ok {
+			// We probably should just preserve behavior in this case.
+			// but not let the problem get worse. having a single one will
+			// suffice.
+			klog.V(2).Infof("snapshot driver: %v has multiple default classes", snapshotClass.Driver)
+		}
+		driverToSnapshotDefautlMap[snapshotClass.Driver] = snapshotClass
+	}
+
+	if _, ok := driverToSnapshotDefautlMap[snapClass.Driver]; ok {
+		reviewResponse.Allowed = false
+		reviewResponse.Result.Message = fmt.Sprintf("default snapshot class already exits for driver: %v", snapClass.Driver)
```

Comment:
Please add the existing VolumeSnapshotClass name to the message, so people don't need to list all storage classes to find the conflicting one.